Database kt

bemusementpark opened the pull request.

Commit:
Convert Database

Commit:
Convert AttachmentDatabase

Commit:
Cleanup AttachmentDatabase

Commit:
Convert DatabaseAttachment

Commit:
Convert MediaDatabase

Commit:
Cleanup MediaDatabase

AL-Session commented:
All seems nicer, just as long as everything still works we could incorporate - but would prolly need a separate regression test past if `on-2` has already passed or in QA? Maybe we could merge into `dev` at some point after `on-1` & `on-2` have gone in?

mpretty-cyro committed:
Updated the geolite2 database, removed the IP from the paths page

mpretty-cyro committed:
Added an attribution label to the PathActivity

mpretty-cyro committed:
Merge branch 'dev' into update-geolite2

mpretty-cyro committed:
Resolving PR comment

ThomasSession committed:
Fix/video call rotation and avatars (#1548)

* Simplifying profile picture view

We don't need the isLarge option as the component's size is always set.
Using profilePictureView in the call screen.

* Swapping avatars between user and contact's

* Adding the user's avatar for when it  needs to be displayed

* Making sure we never invert the contact's landscape rotation

* Update app/src/main/java/org/thoughtcrime/securesms/calls/WebRtcCallActivity.kt

Co-authored-by: Andrew <andrewgallasch@gmail.com>

* Update app/src/main/java/org/thoughtcrime/securesms/calls/WebRtcCallActivity.kt

Co-authored-by: Andrew <andrewgallasch@gmail.com>

* Update app/src/main/java/org/thoughtcrime/securesms/calls/WebRtcCallActivity.kt

Co-authored-by: Andrew <andrewgallasch@gmail.com>

---------

Co-authored-by: Andrew <andrewgallasch@gmail.com>

mpretty-cyro committed:
Merge pull request #1545 from mpretty-cyro/update-geolite2

Updated GeoLite2 database

mpretty-cyro committed:
Removed the attribution label

mpretty-cyro committed:
Merge pull request #1550 from mpretty-cyro/remove-attribution

Remove attribution from path screen

bemusementpark committed:
Merge pull request #1451 from bemusementpark/on-2

[SES-48 SES-824] Onboarding Overhaul

Commit:
Merge branch 'dev' into databaseKt

Commit:
Cleanup

Commit:
Simplify MmsDatabase

Commit:
Improve Database

Commit:
Convert LinkPreview

Commit:
Simplify LinkPreview

Commit:
Simplify MmsDatabase
